sequence_processing_pipeline/NuQCJob.py

```diff
@@ -167,33 +189,69 @@ def run(self, callback=None):
                                    # for now.
                                    exec_from=self.log_path,
                                    callback=callback)
-
         job_id = job_info['job_id']
-        logging.debug(f'QCJob {job_id} completed')
+        logging.debug(f'NuQCJob {job_id} completed')
 
         for project in self.project_data:
             project_name = project['Sample_Project']
             needs_human_filtering = project['HumanFiltering']
 
             source_dir = join(self.output_path, project_name)
+            pattern = f"{source_dir}/*.fastq.gz"
+            completed_files = list(glob.glob(pattern))
 
-            if not self._get_failed_indexes(project_name, job_id):
-                raise PipelineError("QCJob did not complete successfully.")
-
-            # TODO: IMPLEMNENT A NEW FILTER FOR FILTERED FASTQ.GZ FILES THAT
-            #  ARE BELOW THE MINIMUM FILE SIZE THRESHOLD INTO A NEW FOLDER
-            #  NAMED 'ZERO-LENGTH-FILES'.
-
-            # determine where the filtered fastq files can be found and move
-            # the 'zero-length' files to another directory.
             if needs_human_filtering is True:
                 filtered_directory = join(source_dir, 'filtered_sequences')
             else:
                 filtered_directory = join(source_dir, 'trimmed_sequences')
 
-            if not exists(filtered_directory):
-                raise PipelineError(f"{filtered_directory} does not exist")
-
+            # create the properly named directory to move files to in
+            # in order to preserve legacy behavior.
+            makedirs(filtered_directory, exist_ok=True)
+
+            # get the list of sample-names in this project.
+            samples_in_project = [x[0] for x in self.sample_ids
+                                  if x[1] == project_name]
+
+            # Tissue_1_Mag_Hom_DNASe_RIBO_S16_L001_R2_001.fastq.gz
+            # Nislux_SLC_Trizol_DNASe_S7_L001_R2_001.fastq.gz
+            self._move_helper(completed_files,
+                              r'^(.*)_S\d{1,2}_L\d{3}_R\d_\d{3}\.fastq\.gz$',
```

Comment:
Can the regular expressions be pulled out and associated with unit tests that assert expected matching?

Reply:
...instead of passing a pattern to compile, an existing compiled expression could be provided. That then makes it very easy to unit test the expression